src/dispersion_relation.jl

```diff
@@ -0,0 +1,119 @@
+@doc raw"""
+    LinearDispersionRelation(ref_height)
+
+A struct representing a linear dispersion relation ``\omega(k)`` of an equation. The reference
+water height `h0` is given by `ref_height`. A dispersion relation can be called as
+`disp_rel(equations, k)` to compute the wave frequency ``\omega(k)`` for a given wavenumber `k`
+and a set of equations.
+
+See also [`wave_speed`](@ref) for computing the wave speed ``c = \omega(k) / k`` given a linear
+dispersion relation.
+"""
+struct LinearDispersionRelation{RealT <: Real}
+    ref_height::RealT
+end
+
+function Base.show(io::IO, disp_rel::LinearDispersionRelation)
+    print(io, "LinearDispersionRelation(h0 = ", disp_rel.ref_height, ")")
+end
+
+Base.broadcastable(disp_rel::LinearDispersionRelation) = (disp_rel,)
+
+@doc raw"""
+    wave_speed(disp_rel, equations, k; normalize = false)
+
+Compute the wave speed ``c`` for a given wavenumber ``k`` using the
+[`LinearDispersionRelation`](@ref) `disp_rel` of the `equations`.
+The wave speed is given by ``c = \omega(k) / k``. If `normalize` is `true`, the wave speed is normalized
+by the shallow water wave speed ``\sqrt{g h_0}``, where ``g`` is the `gravity_constant` of the `equations`
+and ``h_0`` is the `ref_height` of the dispersion relation `disp_rel`.
+
+See also [`LinearDispersionRelation`](@ref).
+"""
+function wave_speed(disp_rel::LinearDispersionRelation, equations, k;
+                    normalize = false)
+    omega = disp_rel(equations, k)
+    c = omega / k
+    if normalize
+        c /= sqrt(gravity_constant(equations) * disp_rel.ref_height)
+    end
+    return c
+end
+
+@doc raw"""
+    EulerEquations1D(; gravity_constant, eta0 = 0.0)
+
+A struct representing the 1D Euler equations with a given gravity constant and a still-water surface
+`eta0`.
+
+!!! note
+    In DispersiveShallowWater.jl, the Euler equations are *only* used for computing the full linear dispersion
+    relation
+    ```math
+    \omega(k) = \sqrt{g k \tanh(h_0 k)}.
+    ```
+    They cannot be solved as a system of equations.
+"""
```

Comment:
```suggestion
!!! note
    In DispersiveShallowWater.jl, the Euler equations are *only* used for computing
    the full linear dispersion relation
    ```math
    \omega(k) = \sqrt{g k \tanh(h_0 k)},
    ```
    see [`LinearDispersionRelation`](@ref) and [`wave_speed`](@ref).
    The `EulerEquations1D` cannot be solved as a system of equations.
"""
```